patchwork-extensions-keybinds/src/main/java/com/patchworkmc/mixin/extensions/keybinds/MixinKeyBinding.java

```diff
@@ -0,0 +1,206 @@
+import net.minecraft.client.util.InputUtil;
+
+import com.patchworkmc.impl.extensions.keybinds.IPatchworkKeyBinding;
+
+@Mixin(KeyBinding.class)
+public abstract class MixinKeyBinding implements IForgeKeybinding, IPatchworkKeyBinding {
+	@Shadow
+	private InputUtil.KeyCode keyCode;
+
+	@Mutable
+	@Shadow
+	@Final
+	private static Map<InputUtil.KeyCode, KeyBinding> keysByCode;
+
+	@Shadow
+	public abstract String getCategory();
+
+	@Shadow
+	public abstract String getId();
+
+	@Shadow
+	@Final
+	private static Map<String, Integer> categoryOrderMap;
+	private KeyModifier keyModifierDefault = KeyModifier.NONE;
+	private KeyModifier keyModifier = KeyModifier.NONE;
+	private IKeyConflictContext keyConflictContext = KeyConflictContext.UNIVERSAL;
+
+	@Inject(method = "<clinit>", at = @At("RETURN"))
+	private static void staticInit(CallbackInfo ci) {
+		keysByCode = new KeyBindingMap();
+	}
+
+	@Inject(method = "onKeyPressed", at = @At("HEAD"), cancellable = true)
+	private static void onKeyPressed(InputUtil.KeyCode keyCode, CallbackInfo ci) {
+		ci.cancel();
+
+		for (KeyBinding keybinding : ((KeyBindingMap) keysByCode).lookupAll(keyCode)) {
+			if (keybinding != null) {
+				((KeyBindingHooks) keybinding).setTimesPressed(((KeyBindingHooks) keybinding).getTimesPressed() + 1);
+			}
+		}
+	}
+
+	@Inject(method = "setKeyPressed", at = @At("HEAD"), cancellable = true)
+	private static void onKeyPressed(InputUtil.KeyCode keyCode, boolean pressed, CallbackInfo ci) {
+		ci.cancel();
+
+		for (KeyBinding keybinding : ((KeyBindingMap) keysByCode).lookupAll(keyCode)) {
+			if (keybinding != null) {
+				((KeyBindingHooks) keybinding).setPressed(pressed);
+			}
+		}
+	}
+
+	@Inject(method = "isPressed", at = @At("RETURN"), cancellable = true)
+	private void isPressed(CallbackInfoReturnable<Boolean> cir) {
+		cir.setReturnValue(cir.getReturnValueZ() && getKeyConflictContext().isActive() && getKeyModifier().isActive(getKeyConflictContext()));
+	}
+
+	@Inject(method = "method_1430", at = @At("HEAD"), cancellable = true)
+	private void compareTo(KeyBinding keyBinding, CallbackInfoReturnable<Integer> cir) {
+		if (this.getCategory().equals(keyBinding.getCategory())) {
+			cir.setReturnValue(I18n.translate(this.getId()).compareTo(I18n.translate(keyBinding.getId())));
+			return;
+		}
+
+		Integer tCat = categoryOrderMap.get(this.getCategory());
+		Integer oCat = categoryOrderMap.get(keyBinding.getCategory());
+
+		if (tCat == null && oCat != null) {
+			cir.setReturnValue(1);
+			return;
+		}
+
+		if (tCat != null && oCat == null) {
+			cir.setReturnValue(-1);
+			return;
+		}
+
+		if (tCat == null && oCat == null) {
+			cir.setReturnValue(I18n.translate(this.getCategory()).compareTo(I18n.translate(keyBinding.getCategory())));
+			return;
+		}
+
+		cir.setReturnValue(tCat.compareTo(oCat));
+	}
+
+	@Inject(method = "equals", at = @At("HEAD"), cancellable = true)
+	private void equals(KeyBinding binding, CallbackInfoReturnable<Boolean> cir) {
```

Comment:
This function has a lot of repetition of certain fragments of code (such as `((IForgeKeybinding) binding).getKeyConflictContext()`), reducing the line count but making it harder to read. While Forge wants to reduce line count in patches, I think it is best to expand that kind of stuff here.

```suggestion
	private void conflictAwareEquals(KeyBinding binding, CallbackInfoReturnable<Boolean> cir) {
```